Src/DependencyCollector/Shared/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,361 @@
+            writeListField = connectionType?.GetField("m_WriteList", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            // Second step: Generate an accessor for HttpWebRequest._HttpResponse
+            FieldInfo field = typeof(HttpWebRequest).GetField("_HttpResponse", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            string methodName = field?.ReflectedType.FullName + ".get_" + field.Name;
+            if (!string.IsNullOrEmpty(methodName))
+            {
+                DynamicMethod getterMethod = new DynamicMethod(methodName, typeof(HttpWebResponse), new Type[] { typeof(HttpWebRequest) }, true);
+                ILGenerator generator = getterMethod.GetILGenerator();
+                generator.Emit(OpCodes.Ldarg_0);
+                generator.Emit(OpCodes.Ldfld, field);
+                generator.Emit(OpCodes.Ret);
+                httpResponseAccessor = (Func<HttpWebRequest, HttpWebResponse>)getterMethod.CreateDelegate(typeof(Func<HttpWebRequest, HttpWebResponse>));
+            }
+
+            // Double checking to make sure we have all the pieces initialized
+            if (connectionGroupListField == null ||
+                connectionGroupType == null ||
+                connectionListField == null ||
+                connectionType == null ||
+                writeListField == null ||
+                httpResponseAccessor == null)
+            {
+                throw new InvalidOperationException("Unable to initialize all required reflection objects");
+            }
+        }
+
+        private static void PerformInjection()
+        {
+            FieldInfo servicePointTableField = typeof(ServicePointManager).GetField("s_ServicePointTable", BindingFlags.Static | BindingFlags.NonPublic);
+            if (servicePointTableField == null)
+            {
+                throw new InvalidOperationException("Unable to access the ServicePointTable field");
+            }
+
+            Hashtable originalTable = servicePointTableField.GetValue(null) as Hashtable;
+            ServicePointHashtable newTable = new ServicePointHashtable();
+
+            // Copy the existing entries over to the new table, and replace the field with our new table
+            if (originalTable != null)
+            {
+                foreach (DictionaryEntry entry in originalTable)
+                {
+                    newTable[entry.Key] = entry.Value;
+                }
+            }
+
+            servicePointTableField.SetValue(null, newTable);
+        }
+
+        /// <summary>
+        /// Initializes all the reflection objects it will ever need. Reflection is costly, but it's better to take
+        /// this one time performance hit than to get it multiple times later, or do it lazily and have to worry about
+        /// threading issues. If Initialize has been called before, it will not doing anything.
+        /// </summary>
+        private void Initialize()
```

Comment:
This file is temporary. The real implementation will sit in System.Diagnostics.DiagnosticSource, see this review at:
https://github.com/dotnet/corefx/pull/18101

Once it's in DiagnosticSource dll, this file will be removed. It also makes more sense if you look at how we trigger the initialization from DiagnosticListener, it becomes clear that you don't want to use a static constructor because it should get initialized only if someone is subscribing for listening.